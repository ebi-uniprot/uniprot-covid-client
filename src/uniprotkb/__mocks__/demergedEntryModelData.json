{
  "entryType": "Inactive",
  "primaryAccession": "Q00007",
  "uniProtId": "2ABA_HUMAN",
  "annotationScore": 0.0,
  "inactiveReason": {
    "inactiveReasonType": "DEMERGED",
    "mergeDemergeTo": ["P63151", "P63150"]
  }
}