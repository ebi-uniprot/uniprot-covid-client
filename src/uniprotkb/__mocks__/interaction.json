[
  {
    "accession": "P09958",
    "name": "FURIN_HUMAN",
    "proteinExistence": "Evidence at protein level",
    "taxonomy": 9606,
    "interactions": [
      {
        "accession1": "P09958",
        "accession2": "P50281",
        "gene": "MMP14",
        "interactor1": "EBI-1056807",
        "interactor2": "EBI-992788",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P09958",
        "accession2": "Q9H239",
        "gene": "MMP28",
        "interactor1": "EBI-1056807",
        "interactor2": "EBI-20858485",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P09958",
        "accession2": "O14793",
        "gene": "MSTN",
        "interactor1": "EBI-1056807",
        "interactor2": "EBI-8542977",
        "organismDiffer": false,
        "experiments": 2
      }
    ],
    "subcellularLocations": [
      {
        "type": "SUBCELLULAR_LOCATION",
        "locations": [
          {
            "location": {
              "value": "Golgi apparatus, trans-Golgi network membrane",
              "evidences": [
                {
                  "code": "ECO:0000269",
                  "source": {
                    "name": "PubMed",
                    "id": "11331585",
                    "url": "http://www.ncbi.nlm.nih.gov/pubmed/11331585",
                    "alternativeUrl": "https://europepmc.org/abstract/MED/11331585"
                  }
                },
                {
                  "code": "ECO:0000269",
                  "source": {
                    "name": "PubMed",
                    "id": "11799113",
                    "url": "http://www.ncbi.nlm.nih.gov/pubmed/11799113",
                    "alternativeUrl": "https://europepmc.org/abstract/MED/11799113"
                  }
                },
                {
                  "code": "ECO:0000269",
                  "source": {
                    "name": "PubMed",
                    "id": "8846780",
                    "url": "http://www.ncbi.nlm.nih.gov/pubmed/8846780",
                    "alternativeUrl": "https://europepmc.org/abstract/MED/8846780"
                  }
                },
                {
                  "code": "ECO:0000269",
                  "source": {
                    "name": "PubMed",
                    "id": "9130696",
                    "url": "http://www.ncbi.nlm.nih.gov/pubmed/9130696",
                    "alternativeUrl": "https://europepmc.org/abstract/MED/9130696"
                  }
                },
                {
                  "code": "ECO:0000269",
                  "source": {
                    "name": "PubMed",
                    "id": "9412467",
                    "url": "http://www.ncbi.nlm.nih.gov/pubmed/9412467",
                    "alternativeUrl": "https://europepmc.org/abstract/MED/9412467"
                  }
                }
              ]
            },
            "topology": {
              "value": "Single-pass type I membrane protein",
              "evidences": [
                {
                  "code": "ECO:0000305"
                }
              ]
            }
          },
          {
            "location": {
              "value": "Cell membrane",
              "evidences": [
                {
                  "code": "ECO:0000269",
                  "source": {
                    "name": "PubMed",
                    "id": "11799113",
                    "url": "http://www.ncbi.nlm.nih.gov/pubmed/11799113",
                    "alternativeUrl": "https://europepmc.org/abstract/MED/11799113"
                  }
                },
                {
                  "code": "ECO:0000269",
                  "source": {
                    "name": "PubMed",
                    "id": "9130696",
                    "url": "http://www.ncbi.nlm.nih.gov/pubmed/9130696",
                    "alternativeUrl": "https://europepmc.org/abstract/MED/9130696"
                  }
                },
                {
                  "code": "ECO:0000269",
                  "source": {
                    "name": "PubMed",
                    "id": "9412467",
                    "url": "http://www.ncbi.nlm.nih.gov/pubmed/9412467",
                    "alternativeUrl": "https://europepmc.org/abstract/MED/9412467"
                  }
                }
              ]
            },
            "topology": {
              "value": "Single-pass type I membrane protein",
              "evidences": [
                {
                  "code": "ECO:0000305"
                }
              ]
            }
          },
          {
            "location": {
              "value": "Secreted",
              "evidences": [
                {
                  "code": "ECO:0000305",
                  "source": {
                    "name": "PubMed",
                    "id": "11799113",
                    "url": "http://www.ncbi.nlm.nih.gov/pubmed/11799113",
                    "alternativeUrl": "https://europepmc.org/abstract/MED/11799113"
                  }
                }
              ]
            }
          },
          {
            "location": {
              "value": "Endosome membrane",
              "evidences": [
                {
                  "code": "ECO:0000269",
                  "source": {
                    "name": "PubMed",
                    "id": "9412467",
                    "url": "http://www.ncbi.nlm.nih.gov/pubmed/9412467",
                    "alternativeUrl": "https://europepmc.org/abstract/MED/9412467"
                  }
                }
              ]
            },
            "topology": {
              "value": "Single-pass type I membrane protein",
              "evidences": [
                {
                  "code": "ECO:0000305"
                }
              ]
            }
          }
        ],
        "text": [
          {
            "value": "Shuttles between the trans-Golgi network and the cell surface (PubMed:9412467, PubMed:11799113). Propeptide cleavage is a prerequisite for exit of furin molecules out of the endoplasmic reticulum (ER). A second cleavage within the propeptide occurs in the trans Golgi network (TGN), followed by the release of the propeptide and the activation of furin (PubMed:11799113)",
            "evidences": [
              {
                "code": "ECO:0000269",
                "source": {
                  "name": "PubMed",
                  "id": "11799113",
                  "url": "http://www.ncbi.nlm.nih.gov/pubmed/11799113",
                  "alternativeUrl": "https://europepmc.org/abstract/MED/11799113"
                }
              },
              {
                "code": "ECO:0000269",
                "source": {
                  "name": "PubMed",
                  "id": "9412467",
                  "url": "http://www.ncbi.nlm.nih.gov/pubmed/9412467",
                  "alternativeUrl": "https://europepmc.org/abstract/MED/9412467"
                }
              }
            ]
          }
        ]
      }
    ]
  },
  {
    "accession": "O14793",
    "name": "GDF8_HUMAN",
    "proteinExistence": "Evidence at protein level",
    "taxonomy": 9606,
    "interactions": [
      {
        "accession1": "O14793",
        "accession2": "Q13705",
        "gene": "ACVR2B",
        "interactor1": "EBI-8542977",
        "interactor2": "EBI-1383577",
        "organismDiffer": false,
        "experiments": 4
      },
      {
        "accession1": "O14793",
        "accession2": "P13497",
        "gene": "BMP1",
        "interactor1": "EBI-8542977",
        "interactor2": "EBI-489827",
        "organismDiffer": false,
        "experiments": 2
      },
      {
        "accession1": "O14793",
        "accession2": "P09958",
        "gene": "FURIN",
        "interactor1": "EBI-8542977",
        "interactor2": "EBI-1056807",
        "organismDiffer": false,
        "experiments": 2
      },
      {
        "accession1": "O14793",
        "accession2": "O14793",
        "gene": "MSTN",
        "interactor1": "EBI-8542977",
        "interactor2": "EBI-8542977",
        "organismDiffer": false,
        "experiments": 6
      },
      {
        "accession1": "O14793",
        "accession2": "Q96NZ8",
        "gene": "WFIKKN1",
        "interactor1": "EBI-8542977",
        "interactor2": "EBI-2363713",
        "organismDiffer": false,
        "experiments": 4
      },
      {
        "accession1": "O14793",
        "chain1": "PRO_0000033951",
        "accession2": "O14793",
        "chain2": "PRO_0000033950",
        "gene": "MSTN",
        "interactor1": "EBI-20717179",
        "interactor2": "EBI-20717185",
        "organismDiffer": false,
        "experiments": 3
      }
    ],
    "diseases": [
      {
        "type": "DISEASE",
        "diseaseId": "Muscle hypertrophy",
        "acronym": "MSLHP",
        "dbReference": {
          "type": "MIM",
          "id": "614160"
        },
        "description": {
          "value": "A condition characterized by increased muscle bulk and strength. Affected individuals are exceptionally strong.",
          "evidences": [
            {
              "code": "ECO:0000269",
              "source": {
                "name": "PubMed",
                "id": "15215484",
                "url": "http://www.ncbi.nlm.nih.gov/pubmed/15215484",
                "alternativeUrl": "https://europepmc.org/abstract/MED/15215484"
              }
            }
          ]
        },
        "text": [
          {
            "value": "The disease is caused by mutations affecting the gene represented in this entry"
          }
        ]
      }
    ],
    "subcellularLocations": [
      {
        "type": "SUBCELLULAR_LOCATION",
        "locations": [
          {
            "location": {
              "value": "Secreted",
              "evidences": [
                {
                  "code": "ECO:0000250",
                  "source": {
                    "name": "UniProtKB",
                    "id": "O08689",
                    "url": "https://www.uniprot.org/uniprot/O08689"
                  }
                }
              ]
            }
          }
        ]
      }
    ]
  },
  {
    "accession": "Q9H239",
    "name": "MMP28_HUMAN",
    "proteinExistence": "Evidence at protein level",
    "taxonomy": 9606,
    "interactions": [
      {
        "accession1": "Q9H239",
        "accession2": "P09958",
        "gene": "FURIN",
        "interactor1": "EBI-20858485",
        "interactor2": "EBI-1056807",
        "organismDiffer": false,
        "experiments": 3
      }
    ],
    "subcellularLocations": [
      {
        "type": "SUBCELLULAR_LOCATION",
        "locations": [
          {
            "location": {
              "value": "Secreted, extracellular space, extracellular matrix"
            }
          }
        ]
      }
    ]
  },
  {
    "accession": "P50281",
    "name": "MMP14_HUMAN",
    "proteinExistence": "Evidence at protein level",
    "taxonomy": 9606,
    "interactions": [
      {
        "accession1": "P50281",
        "accession2": "Q9BV57",
        "gene": "ADI1",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-992807",
        "organismDiffer": false,
        "experiments": 4
      },
      {
        "accession1": "P50281",
        "accession2": "Q13520",
        "gene": "AQP6",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-13059134",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "P56945",
        "gene": "BCAR1",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-702093",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "Q8N6L0",
        "gene": "CCDC155",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-749265",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "Q15125",
        "gene": "EBP",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-3915253",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "Q9GZR5",
        "gene": "ELOVL4",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-18535450",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "Q9Y624",
        "gene": "F11R",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-742600",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "P22455",
        "gene": "FGFR4",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-6256193",
        "organismDiffer": false,
        "experiments": 6
      },
      {
        "accession1": "P50281",
        "accession2": "Q13643",
        "gene": "FHL3",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-741101",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "P09958",
        "gene": "FURIN",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-1056807",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "Q7Z5P4",
        "gene": "HSD17B13",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-18053395",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "P53667",
        "gene": "LIMK1",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-444403",
        "organismDiffer": false,
        "experiments": 4
      },
      {
        "accession1": "P50281",
        "accession2": "P51884",
        "gene": "LUM",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-725780",
        "organismDiffer": false,
        "experiments": 2
      },
      {
        "accession1": "P50281",
        "accession2": "Q15800",
        "gene": "MSMO1",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-949102",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "P27105",
        "gene": "STOM",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-1211440",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "P16035",
        "gene": "TIMP2",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-1033507",
        "organismDiffer": false,
        "experiments": 2
      },
      {
        "accession1": "P50281",
        "accession2": "Q96AP4",
        "gene": "ZUP1",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-744706",
        "organismDiffer": false,
        "experiments": 3
      },
      {
        "accession1": "P50281",
        "accession2": "Q9R064",
        "gene": "Gorasp2",
        "interactor1": "EBI-992788",
        "interactor2": "EBI-4422912",
        "organismDiffer": true,
        "experiments": 14
      }
    ],
    "diseases": [
      {
        "type": "DISEASE",
        "diseaseId": "Winchester syndrome",
        "acronym": "WNCHRS",
        "dbReference": {
          "type": "MIM",
          "id": "277950"
        },
        "description": {
          "value": "A disease characterized by severe osteolysis in the hands and feet, generalized osteoporosis, bone thinning, and absence of subcutaneous nodules. Various additional features include coarse face, corneal opacities, gum hypertrophy, and EKG changes.",
          "evidences": [
            {
              "code": "ECO:0000269",
              "source": {
                "name": "PubMed",
                "id": "22922033",
                "url": "http://www.ncbi.nlm.nih.gov/pubmed/22922033",
                "alternativeUrl": "https://europepmc.org/abstract/MED/22922033"
              }
            }
          ]
        },
        "text": [
          {
            "value": "The disease is caused by mutations affecting the gene represented in this entry"
          }
        ]
      }
    ],
    "subcellularLocations": [
      {
        "type": "SUBCELLULAR_LOCATION",
        "locations": [
          {
            "location": {
              "value": "Membrane",
              "evidences": [
                {
                  "code": "ECO:0000305"
                }
              ]
            },
            "topology": {
              "value": "Single-pass type I membrane protein",
              "evidences": [
                {
                  "code": "ECO:0000305"
                }
              ]
            }
          },
          {
            "location": {
              "value": "Melanosome"
            }
          },
          {
            "location": {
              "value": "Cytoplasm"
            }
          }
        ],
        "text": [
          {
            "value": "Identified by mass spectrometry in melanosome fractions from stage I to stage IV. Forms a complex with BST2 and localizes to the cytoplasm"
          }
        ]
      }
    ]
  }
]
