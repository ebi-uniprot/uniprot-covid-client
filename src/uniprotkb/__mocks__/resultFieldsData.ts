import apiUrls from '../config/apiUrls';

const mockResultFieldsApi = {
  request: apiUrls.resultsFields,
  response: [
    {
      groupName: 'Names & Taxonomy',
      fields: [
        {
          label: 'Entry',
          name: 'accession',
        },
        {
          label: 'Entry Name',
          name: 'id',
        },
        {
          label: 'Gene Names',
          name: 'gene_names',
        },
        {
          label: 'Gene Names (ordered locus)',
          name: 'gene_oln',
        },
      ],
    },
    {
      groupName: 'Sequence',
      fields: [
        {
          label: 'EMBL',
          name: 'dr:embl',
        },
        {
          label: 'CCDS',
          name: 'dr:ccds',
        },
        {
          label: 'PIR',
          name: 'dr:pir',
        },
        {
          label: 'RefSeq',
          name: 'dr:refseq',
        },
      ],
      isDatabase: true,
    },
  ],
};

export default mockResultFieldsApi;
