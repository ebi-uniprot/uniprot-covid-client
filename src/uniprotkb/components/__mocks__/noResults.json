{
  "facets": [{
    "label": "Status",
    "name": "reviewed",
    "allowMultipleSelection": false
  }, {
    "label": "Popular organisms",
    "name": "popular_organism",
    "allowMultipleSelection": true
  }, {
    "label": "Proteins with",
    "name": "proteins_with",
    "allowMultipleSelection": true
  }, {
    "label": "Protein Existence",
    "name": "existence",
    "allowMultipleSelection": true
  }, {
    "label": "Annotation Score",
    "name": "annotation_score",
    "allowMultipleSelection": true
  }, {
    "label": "Sequence length",
    "name": "length",
    "allowMultipleSelection": true,
    "values": [{
      "label": "1 - 200",
      "value": "[1 TO 200]",
      "count": 0
    }, {
      "label": "201 - 400",
      "value": "[201 TO 400]",
      "count": 0
    }, {
      "label": "401 - 600",
      "value": "[401 TO 600]",
      "count": 0
    }, {
      "label": "601 - 800",
      "value": "[601 TO 800]",
      "count": 0
    }, {
      "label": ">= 801",
      "value": "[801 TO *]",
      "count": 0
    }]
  }],
  "matchedFields": [],
  "results": []
}