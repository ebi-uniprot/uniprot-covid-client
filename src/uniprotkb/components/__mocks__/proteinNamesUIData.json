{
  "recommendedName": "My protein",
  "shortNames": "some shorter names",
  "alternativeNames": ["alt1", "alt2"]
}
