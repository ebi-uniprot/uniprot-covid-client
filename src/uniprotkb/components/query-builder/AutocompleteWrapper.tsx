import React, { Component } from 'react';
import { Autocomplete } from 'franklin-sites';
import { v1 } from 'uuid';
import fetchData from '../../../shared/utils/fetchData';
import { getSuggesterUrl } from '../../config/apiUrls';

type Props = {
  url: string;
  onSelect: (path: string, id?: string) => void;
  title: string;
  placeholder?: string;
  value?: string;
  clearOnSelect?: boolean;
};

type Suggestion = {
  value: string;
  id: string;
};

export type Suggestions = {
  dictionary: string;
  query: string;
  suggestions: Suggestion[];
};

type State = {
  data: SelectValue[];
  previousTextInputValue: string;
  isLoading: boolean;
};

type SelectValue = {
  id: string;
  itemLabel: string;
  pathLabel: string;
  apiId: string;
};

const minCharsToShowDropdown = 3;

class AutocompleteWrapper extends Component<Props, State> {
  static prepareData(suggestions: Suggestion[]) {
    return suggestions.map(
      (suggestion: Suggestion): SelectValue => ({
        pathLabel: `${suggestion.value} [${suggestion.id}]`,
        itemLabel: suggestion.value,
        apiId: suggestion.id,
        id: v1(),
      })
    );
  }

  id: string;

  constructor(props: Props) {
    super(props);
    this.state = { data: [], previousTextInputValue: '', isLoading: false };
    this.id = v1();
  }

  handleChange = (textInputValue: string) => {
    this.handleSelect(textInputValue);
    const { previousTextInputValue } = this.state;
    const trimmedTextInputValue = textInputValue.trim();
    // If there aren't enough characters then reset the state's data
    // and don't fetch suggestions.
    if (trimmedTextInputValue.trim().length < minCharsToShowDropdown) {
      this.setState({
        data: [],
        previousTextInputValue: trimmedTextInputValue,
      });
    }
    // If the previous text input and the current text input are
    // subsets of each other then don't reset the state's data. The
    // point of this check is to prevent the dropdown disappearing
    // and reappearing between successive suggester requests.
    // Fetch new suggestions after previousTextInputValue has been
    // set. Use the callback to prevent race conditions.
    else if (
      trimmedTextInputValue.includes(previousTextInputValue) ||
      previousTextInputValue.includes(trimmedTextInputValue)
    ) {
      this.setState({ previousTextInputValue: trimmedTextInputValue }, () =>
        this.fetchOptions(trimmedTextInputValue)
      );
    }
    // This is a "new" or "unrelated" text input so erase the old data
    // and then fetch the new suggestions after the state has been set.
    // Use the callback to prevent race conditions.
    else {
      this.setState(
        {
          data: [],
          previousTextInputValue: trimmedTextInputValue,
        },
        () => this.fetchOptions(trimmedTextInputValue)
      );
    }
  };

  fetchOptions = (textInputValue: string) => {
    const { url } = this.props;
    const suggesterUrl = getSuggesterUrl(url, textInputValue);
    this.setState({ isLoading: true });
    fetchData<Suggestions>(suggesterUrl)
      .then((response) =>
        AutocompleteWrapper.prepareData(response.data.suggestions)
      )
      .then((data) => this.setState({ data, isLoading: false }))
      // eslint-disable-next-line no-console
      .catch((e) => console.error(e));
  };

  handleSelect = (inputValue: SelectValue | string) => {
    const { onSelect } = this.props;
    if (typeof inputValue === 'string') {
      onSelect(inputValue);
    } else {
      onSelect(inputValue.pathLabel, inputValue.apiId);
    }
  };

  render() {
    const { data, isLoading } = this.state;
    const {
      title,
      value,
      clearOnSelect = false,
      placeholder = '',
    } = this.props;
    return (
      <label htmlFor={this.id}>
        {title}
        <Autocomplete
          id={this.id}
          data={data}
          onSelect={this.handleSelect}
          onChange={this.handleChange}
          filter={false}
          value={value}
          placeholder={placeholder}
          minCharsToShowDropdown={minCharsToShowDropdown}
          clearOnSelect={clearOnSelect}
          isLoading={isLoading}
        />
      </label>
    );
  }
}

export default AutocompleteWrapper;
