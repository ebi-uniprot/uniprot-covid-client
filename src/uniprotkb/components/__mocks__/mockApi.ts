import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import results from './results.json';
import noResults from './noResults.json';
import entry from './swissprotEntry.json';
import resultFields from '../../__mocks__/resultFields.json';
import mockFasta from './fasta.json';

const mock = new MockAdapter(axios);
mock.onGet(/.+noresult/).reply(200, noResults, { 'x-total-records': 25 });
mock
  .onGet(/\/uniprotkb\/search/)
  .reply(200, results, { 'x-total-records': 25 });
mock.onGet(/\/uniprotkb\/result-fields/).reply(200, resultFields);
mock
  .onGet(
    /\/uniprotkb\/download\?compressed=true&format=fasta&query=nod2&size=10/
  )
  .reply(200, mockFasta, { 'content-type': 'text/fasta' });
mock.onGet(/\/uniprotkb\//).reply(200, entry);

export default mock;
