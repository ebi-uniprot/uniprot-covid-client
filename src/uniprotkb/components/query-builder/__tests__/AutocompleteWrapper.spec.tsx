import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import AutocompleteWrapper from '../AutocompleteWrapper';
import { resetUuidV1 } from '../../../../../__mocks__/uuid';
import {
  mockSuggesterApi,
  preparedSuggestions,
} from './__mocks__/autocompleteWrapperData';

describe('Autocomplete Wrapper static methods', () => {
  test('should prepare API data for Autocomplete', () => {
    resetUuidV1();
    expect(
      AutocompleteWrapper.prepareData(mockSuggesterApi.response.suggestions)
    ).toEqual(preparedSuggestions);
  });
});

const props = {
  title: 'Taxonomy [OC]',
  value: 'Homo sapiens (Human) [9606]',
  inputValue: mockSuggesterApi.query,
  url: mockSuggesterApi.baseUrl,
  onSelect: jest.fn(),
};
const mock = new MockAdapter(axios);
mock.onGet(mockSuggesterApi.request).reply(200, mockSuggesterApi.response);

let rendered;
describe('Autocomplete Wrapper', () => {
  beforeEach(() => {
    resetUuidV1();
    rendered = render(<AutocompleteWrapper {...props} />);
  });

  test('should render', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render the correct number of AutocompleteItems when input is human', async () => {
    const { queryByTestId, getAllByTestId } = rendered;
    const searchInput = queryByTestId('search-input');
    fireEvent.change(searchInput, {
      target: { value: mockSuggesterApi.query },
    });
    const autocompleteItems = await waitFor(() =>
      getAllByTestId('autocomplete-item')
    );
    expect(autocompleteItems.length).toEqual(
      mockSuggesterApi.response.suggestions.length
    );
  });

  test('should not render AutocompleteItems when input is less than minCharsToShowDropdown (=3)', async () => {
    const { queryByTestId, queryAllByTestId } = rendered;
    const searchInput = queryByTestId('search-input');
    const value = 'hu';
    fireEvent.change(searchInput, { target: { value } });
    const autocompleteItems = await waitFor(() =>
      queryAllByTestId('autocomplete-item')
    );
    expect(autocompleteItems.length).toEqual(0);
  });
});
