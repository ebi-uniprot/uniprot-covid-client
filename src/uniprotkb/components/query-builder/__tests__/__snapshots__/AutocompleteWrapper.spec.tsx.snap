// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete Wrapper should render 1`] = `
<DocumentFragment>
  <label
    for="0"
  >
    Taxonomy [OC]
    <div
      class="autocomplete-container"
    >
      <span
        class="search-input"
      >
        <input
          data-testid="search-input"
          placeholder=""
          type="text"
          value="Homo sapiens (Human) [9606]"
        />
        <span
          class="search-input__suffix"
          data-testid="search-input-suffix"
          role="presentation"
        >
          <svg
            fill="currentColor"
            height="14"
            viewBox="0 0 512 512"
            width="14"
          >
            <path
              d="M505 443L405 343c-4-4-10-7-17-7h-16a208 208 0 10-36 36v16c0 7 3 13 7 17l100 100c9 9 24 9 34 0l28-28c9-10 9-25 0-34zM208 336a128 128 0 110-256 128 128 0 010 256z"
            />
          </svg>
        </span>
      </span>
      <div
        class="autocomplete-menu"
      >
        <div
          class="dropdown-menu__panel"
        />
      </div>
    </div>
  </label>
</DocumentFragment>
`;
